apache/superset
# chore: make TS enums strictly PascalCase

### SUMMARY
TypeScript enum names and keys should always be PascalCase, which is the commonly accepted best practice (see the official docs: https://www.typescriptlang.org/docs/handbook/enums.html). This PR does the following:
- Adds ESLint rules to enforce both PascalCase for the enum names and their respective keys.
- Fix the whole FS codebase, including `superset-ui` packages/plugins. The enum values have been left in tact so as to not cause a breaking change.

Note, that I've made a few exceptions, where deviating from pascal case makes more sense, especially the locales and currencies, where it felt dangerous and less intuitive to move to pascal case.

### TESTING INSTRUCTIONS
1. Pull the codebase, try to rename a TS enum to snake case, and notice how the linter complains.

### ADDITIONAL INFORMATION
